### PR TITLE
 Added checks for price format during pluging install and during init…

### PR DESCRIPTION
**Pull request by tmeliotpg**
… Added checks for price format during pluging install and during plugin init in case of setting changes - git rebase

### CODE REVIEW

File: WC_Pagantis.php [hunk 1]
```diff
@@ -315,6 +337,43 @@ public function pagantisActivation()
         update_option('woocommerce_pagantis_settings', $settings);
     }
 
+    /**
+     * Checks the WC settings to know if we should modify our config
+     */
+    public function check_wc_price_settings()
```
Comment (rmarinleal) [inline WC_Pagantis.php]: Is not possible to preserve the camelCase in these function names?

File: includes/pg-functions.php [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    $tableName = $wpdb->prefix . PG_LOGS_TABLE_NAME;
+    $wpdb->insert($tableName, array('log' => $logEntry->toJson()));
+}
+
+/**
+ * @return bool
+ */
+function areDecimalSeparatorEqual()
+{
+    $decimalSeparator                = getPgSimulatorDecimalSeparatorConfig();
```
Comment (rmarinleal) [inline includes/pg-functions.php]: Try not tu use unnecesary blank spaces ($var = 'val'; )

File: includes/pg-functions.php [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+/**
+ * @return array|object|null
+ */
+function getPgSimulatorDecimalSeparatorConfig()
+{
+    global $wpdb;
+    $tableName = $wpdb->prefix . PG_CONFIG_TABLE_NAME;
+    $query     = "SELECT * FROM $tableName WHERE config='PAGANTIS_SIMULATOR_DECIMAL_SEPARATOR'";
```
Comment (rmarinleal) [inline includes/pg-functions.php]: try not to use "select * ..."

Comment (rmarinleal) [reply includes/pg-functions.php]: This method returns an arry/object and is compared with 
```
$wc_get_number_of_price_decimals = get_option('woocommerce_price_decimal_sep');
```

Is correct? scalar values is a better option in this case I think

**Comment by tmeliotpg** [comment]
I'm confused about the " try not to use "select * ...""

File: includes/pg-functions.php (hunk 3 above)
Comment (tmeliotpg) [reply includes/pg-functions.php]: > try not to use "select * ..."

~~I'm confused about this suggestion~~
I understand I will fix it

Comment (tmeliotpg) [reply includes/pg-functions.php]: > `$wc_get_number_of_price_decimals = get_option('woocommerce_price_decimal_sep');`

get_option('woocommerce_price_decimal_sep') returns  "."
[see test](https://github.com/woocommerce/woocommerce/blob/master/tests/legacy/unit-tests/formatting/functions.php#L551)
I also dumped the value in the wc-logs with my logging class to double check

File: includes/pg-functions.php (hunk 2 above)
Comment (tmeliotpg) [reply includes/pg-functions.php]: will correct this

File: includes/pg-functions.php (hunk 3 above)
Comment (rmarinleal) [reply includes/pg-functions.php]: so you're comparin an array with a string? on line 79
```
    $thousandSeparator               = getPgSimulatorThousandsSeparator(); -> returns array/object
    $wc_get_price_thousand_separator = get_option('woocommerce_price_thousand_sep'); -> returns string

     if ($wc_get_price_thousand_separator == $thousandSeparator) {
```

File: includes/pg-functions.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    $wpdb->insert($tableName, array('log' => $logEntry->toJson()));
+}
+
+/**
+ * @return bool
+ */
+function areDecimalSeparatorEqual()
+{
+    $pgDecimalSeparator = getPgSimulatorDecimalSeparatorConfig();
+    $wc_decimal_sep= stripslashes(get_option('woocommerce_price_decimal_sep'));
```
Comment (rmarinleal) [inline includes/pg-functions.php]: double stripslashes

File: includes/pg-functions.php [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+
+/**
+ * @return bool
+ */
+function areThousandsSeparatorEqual()
+{
+    $pgThousandSeparator = getPgSimulatorThousandsSeparator();
+    $wc_price_thousand = stripslashes(get_option('woocommerce_price_thousand_sep'));
+    if (stripslashes($wc_price_thousand)== stripslashes($pgThousandSeparator)) {
```
Comment (rmarinleal) [inline includes/pg-functions.php:75]: double stripslashes

Comment (tmeliotpg) [reply includes/pg-functions.php:75]: Removed in following commit

File: includes/pg-functions.php (hunk 4 above)
Comment (tmeliotpg) [reply includes/pg-functions.php]: Removed in following commit

File: WC_Pagantis.php [hunk 6]
```diff
@@ -315,6 +337,43 @@ public function pagantisActivation()
         update_option('woocommerce_pagantis_settings', $settings);
     }
 
+    /**
+     * Checks the WC settings to know if we should modify our config
+     */
+    public function checkWcPriceSettings()
+    {
+        if (! is_product() || ! is_shop()) {
```
Comment (pgarcess) [inline WC_Pagantis.php]: psr2?

Comment (pgarcess) [inline WC_Pagantis.php]: Is the function is_product nad is_shop defined by woocommerce? I can't find in our code

Comment (tmeliotpg) [reply WC_Pagantis.php]: Yes 
[is_product](https://docs.woocommerce.com/wc-apidocs/source-function-is_product.html#81)
[is_shop](https://docs.woocommerce.com/wc-apidocs/source-function-is_product.html#31)

**Review by pgarcess — APPROVED** [review]

File: includes/pg-functions.php (hunk 3 above)
Comment (tmeliotpg) [reply includes/pg-functions.php]: This has been resolved [in this commit](https://github.com/pagantis/woocommerce/pull/55/commits/8e2cd90c1b5cada4a12683b2d4205301bc28e1d6)

File: WC_Pagantis.php (hunk 6 above)
Comment (rmarinleal) [reply WC_Pagantis.php]: Ok!

File: includes/pg-functions.php (hunk 3 above)
Comment (rmarinleal) [reply includes/pg-functions.php]: ok!

**Review by rmarinleal — APPROVED** [review]

**Pull request merged**